src/test/tcp/CMakeLists.txt

```diff
@@ -1,9 +1,19 @@
+find_package(GLIB REQUIRED)
+include_directories(${GLIB_INCLUDES})
+link_libraries(${GLIB_LIBRARIES})
+
 ## build the test as a dynamic executable that plugs into shadow
 add_shadow_plugin(shadow-plugin-test-tcp test_tcp.c)
 
 ## create and install an executable that can run outside of shadow
 add_executable(test-tcp test_tcp.c)
 
+## add mqueue library
```

Comment:
I tried with a fifo like you did for UDP. And as you mentioned in your PR this didn't work ^^
To be honest, I chose this implementation because I had already used it, without thinking about Posix or System V. Your point on the command line it's really good I will check it :)